src/CodeGen_Hexagon.cpp

```diff
@@ -549,14 +549,24 @@ Value *CodeGen_Hexagon::slice_vector(Value *vec, int start, int size) {
     // If we're getting a native vector bits worth of data from half
     // of the argument, we might be able to use lo/hi if the start is appropriate.
     if (size*2 == vec_elements && element_bits*size == native_vector_bits()) {
-        CallInst *call = dyn_cast<CallInst>(vec);
+        CallInst *call;
+        if (BitCastInst *bc = dyn_cast<BitCastInst>(vec)) {
+            call = dyn_cast<CallInst>(bc->getOperand(0));
+        } else {
+            call = dyn_cast<CallInst>(vec);
+        }
         llvm::Function *combine_fn =
             Intrinsic::getDeclaration(module.get(), IPICK(Intrinsic::hexagon_V6_vcombine));
-        if (call && call->getCalledFunction() == combine_fn) {
+        llvm::Type *ret_ty = llvm::VectorType::get(vec_ty->getScalarType(), size);
+        bool is_vcombine = call && call->getCalledFunction() == combine_fn;
+        if (is_vcombine) {
             if (start == 0) {
-                return call->getArgOperand(1);
+                // Not bitcasting here puts the onus on the caller to use call_intrin_cast before
```

Comment:
Makes sense. The reason I had put a comment there was because in a previous incarnation, this code diddn't have the bitcast. Your explanation makes sense, though. I'll remove the comment.